src/hooks/useUserReputation.ts

```diff
@@ -0,0 +1,53 @@
+import { gql, useQuery } from '@apollo/client';
+import { Id } from '@colony/colony-js';
+
+import { Address } from '~types';
+import { ADDRESS_ZERO } from '~constants';
+import { getUserReputation } from '~gql';
+
+interface UseUserReputationHook {
+  userReputation?: string;
+  totalReputation?: string;
+}
+
+const useUserReputation = (
+  colonyAddress?: Address,
+  walletAddress?: Address,
+  domainId = Id.RootDomain,
+  rootHash?: string,
+): UseUserReputationHook => {
+  const { data: userReputationData } = useQuery(gql(getUserReputation), {
+    variables: {
+      input: {
+        colonyAddress: colonyAddress ?? '',
+        walletAddress: walletAddress ?? '',
+        domainId,
+        rootHash,
+      },
+    },
+    fetchPolicy: 'cache-and-network',
+    skip: !colonyAddress || !walletAddress,
```

Comment:
I never noticed the use of `skip` when I first reviewed this. Nice addition :+1:

Reply:
I only added it in one of the latest commits as part of moving it to its own hook